datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/WeaklyTypedAspectsResolver.java

```diff
@@ -74,12 +74,13 @@ public CompletableFuture<List<RawAspect>> get(DataFetchingEnvironment environmen
 
                     DataMap renderSpec = aspectSpec.getRenderSpec();
 
-                    AspectRenderSpec resultRenderSpec = new AspectRenderSpec();
-
-                    resultRenderSpec.setDisplayType(renderSpec.getString("displayType"));
-                    resultRenderSpec.setDisplayName(renderSpec.getString("displayName"));
-                    resultRenderSpec.setKey(renderSpec.getString("key"));
-                    result.setRenderSpec(resultRenderSpec);
+                    if (renderSpec != null) {
+                        AspectRenderSpec resultRenderSpec = new AspectRenderSpec();
+                        resultRenderSpec.setDisplayType(renderSpec.getString("displayType"));
+                        resultRenderSpec.setDisplayName(renderSpec.getString("displayName"));
+                        resultRenderSpec.setKey(renderSpec.getString("key"));
+                        result.setRenderSpec(resultRenderSpec);
+                    }
 
                     results.add(result);
```

Comment:
is the correct behavior when the renderSpec is null to still return the result in the list of results? or should this be in the null check?

Reply:
yup I believe it's fair that it returns null for this piece if there is no renderSpec. It's already an optional field in case the aspect doesn't have a renderSpec

Reply:
Okay great thanks!